mlsdbtool/mcc_update.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+# This will take the mcc data from the file "data/MCC" in the repository
+# tzdata-timed ( https://github.com/sailfishos/tzdata-timed ) and use it
+# to update the code in this repository.
+
+declare -i mcc
+HEADER="../common/mccmapping.h"
+SCRIPT="./mccmapping"
+mcc_file=$1
+if [ -z "$mcc_file" ] ; then
+    echo "Usage: $0 [MCC file]" >&2
+    exit 1
+fi
+if [ ! -e $HEADER ] ; then
+    echo "ERROR: Unable to locate header $HEADER" >&2
+    exit 1
+fi
+if [ ! -e $SCRIPT ] ; then
+    echo "ERROR: Unable to locate script file $SCRIPT" >&2
+    exit 1
+fi
+
+mccs=`cat $mcc_file | grep -v ^# | cut -d ' ' -f1 | sort -nu | tr '\n' ',' | sed 's/^,//;s/,$/,1024/'`
+count=`echo -n $mccs | tr -dc ',' | wc -c`
+
+if [ $count -gt 255 ] ; then
+    echo -e "We have a problem. The functionality of mlsdbtools assumes there are fewer than\n256 unique mccs. The file provided gives us $count. Some manual intervention is needed." >&2
+    exit 1
+fi
+
+echo "Updating $HEADER"
+sed -i "s/=.*/= \{$mccs\}\;/;s/... unique/$count unique/" $HEADER
+
+echo "Updating $SCRIPT"
+last_mcc=0
+first=0
+echo "# MCC -> country code mapping from tzdata-timed repository data/MCC" > $SCRIPT
+for element in `cat $mcc_file | grep -v ^# | cut -d ' ' -f1,3 | sort -u` ; do
+    mcc=$element
+    if [ $mcc -gt 0 ] ; then
+        if [ $mcc -eq $last_mcc ] ; then
```

Comment:
shellcheck'd:
```
In mcc_update.sh line 42:
        if [ $mcc -eq $last_mcc ] ; then
             ^--^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
        if [ "$mcc" -eq $last_mcc ] ; then